lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -4,50 +4,38 @@ class Pod < KubernetesResource
     TIMEOUT = 10.minutes
     SUSPICIOUS_CONTAINER_STATES = %w(ImagePullBackOff RunContainerError ErrImagePull).freeze
 
-    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+    def initialize(name:, namespace:, context:, file: nil, parent: nil, logger:, deploy_started: nil)
       @name = name
       @namespace = namespace
       @context = context
       @file = file
       @parent = parent
+      @deploy_started = deploy_started
```

Comment:
It's not actually new. I added it as a parameter to save the parent resource from having to call the setter immediately in these cases. I'll change it back if you think this introduces confusion. See [here in the diff](https://github.com/Shopify/kubernetes-deploy/pull/107/files#diff-4709b5257970978173e422904663525fL32) for the setter being used instead.

As for why it is needed, it is primarily so that they know when to timeout, although we also use it to filter logs and events. It's not the global deploy start time; it is the time that specific resource was `apply`'d. For resources in your yamls, the instance is created when we first read in the templates, and this attribute is it is set later,  [here](https://github.com/Shopify/kubernetes-deploy/blob/b61fe310824a17ce82f57964bc44d641ffc6c169/lib/kubernetes-deploy/runner.rb#L376).